website_snippet_barcode/static/src/js/website_snippet_barcode.js

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html). */
+
+odoo.define('website_snippet_barcode', function (require) {
+    "use strict";
+
+    var animation = require('web_editor.snippets.animation');
+
+    animation.registry.barcode = animation.Class.extend({
+        selector: '.s_barcode',
+
+        defaults: {
+            type: 'QR',
+            value: encodeURIComponent(window.location.href),
+            aspectratio: '1',
+            width: '500',
+            humanreadable: 'false'
+        },
+
+        start: function () {
+            this.defaults.width = this.$target.width();
+            this.renderBarcode();
+            return this._super();
+        },
+
+        renderBarcode: function () {
+            var options = _.defaults(this._getOptions(this.$target), this.defaults);
+            var height = Math.round(options.width / options.aspectratio);
+            var barcodeSrc = '/report/barcode?type=' + options.type +
```

Comment:
I think template strings (template literals in ES6) would be better here:

```javascript
var barcodeSrc = `/report/barcode`
    + `?type=${ options.type }`
    + `&value=${ options.value }`
    + `&width=${ options.width }`
    + `&height=${ options.height }`
    + `&humanreadable=${ options.humanreadable }`;
```

Which would equal:
```
"/report/barcode?type=QR&value=https%3A%2F%2Fdeveloper.mozilla.org%2Fen-US%2Fdocs%2FWeb%2FJavaScript%2FReference%2FTemplate_literals&width=500&height=undefined&humanreadable=false"
```

Reply:
Bah nvm Phantom probably won't support this

Reply:
😭